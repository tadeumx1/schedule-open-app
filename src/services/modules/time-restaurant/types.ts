export interface ScheduleResponse {
  monday: HourDay[];
  tuesday: HourDay[];
  wednesday: HourDay[];
  thursday: HourDay[];
  friday: HourDay[];
  saturday: HourDay[];
  sunday: HourDay[];
}

export interface HourDay {
  type?: string;
  value?: number;
}
