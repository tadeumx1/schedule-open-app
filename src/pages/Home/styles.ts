import styled from 'styled-components/native';
import { SafeAreaView } from 'react-native-safe-area-context';

export const Container = styled(SafeAreaView)`
  flex: 1;
  background-color: #eeeeee;
  justify-content: center;
  align-items: center;
`;

export const TitleErrorDescription = styled.Text`
  font-family: 'RobotoBold';
  color: #202125;
  font-size: 24px;
`;

export const LoadingHourData = styled.ActivityIndicator.attrs({
  size: 'large',
  color: '#000000',
})``;
