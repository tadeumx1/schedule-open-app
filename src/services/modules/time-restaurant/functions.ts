import { AxiosResponse } from 'axios';
import { api } from '../../client';
import { ScheduleResponse } from './types';

export const getRestaurantScheduleData =
  async (): Promise<ScheduleResponse> => {
    return api
      .get('/scheduleRestaurantWeekData')
      .then((response: AxiosResponse<ScheduleResponse>) => {
        const { data } = response;

        return data;
      })
      .catch((error) => {
        throw new Error(error);
      });
  };
