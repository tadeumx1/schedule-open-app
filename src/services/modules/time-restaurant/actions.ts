import { useQuery } from '@tanstack/react-query';

import { getRestaurantScheduleData } from './functions';

export const GET_RESTAURANT_SCHEDULE_DATA = 'getRestaurantScheduleData';

export const useGetRestaurantScheduleData = () =>
  useQuery([GET_RESTAURANT_SCHEDULE_DATA], getRestaurantScheduleData);
