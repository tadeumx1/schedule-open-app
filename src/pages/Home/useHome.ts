import { useCallback, useEffect, useState } from 'react';
import { useGetRestaurantScheduleData } from '../../services/modules/time-restaurant';
import {
  ScheduleResponse,
  HourDay,
} from '../../services/modules/time-restaurant/types';

interface DayWeekHour {
  open: number;
  closed: number;
}

interface DayWeekHourDescription {
  open: string;
  closed: string;
}

interface DayRestaurantWeek {
  day: string;
  dayTime: DayWeekHour;
}

interface DayRestaurantWeekString {
  day: string;
  dayTime: DayWeekHourDescription;
}

export interface DayRestaurantWeekFormat {
  day: string;
  dayTime: DayWeekHourDescription[];
}

export const useHome = () => {
  const [hourRestaurantData, setHourRestaurantData] = useState<
    DayRestaurantWeekFormat[] | null
  >(null);

  const { data, isError, isSuccess, isLoading } =
    useGetRestaurantScheduleData();

  const handleScheduleRestaurantData = useCallback(
    (dataScheduleResponse: ScheduleResponse): DayRestaurantWeek[] => {
      // Basically transform from a object to array that each day is an array that contains
      // the name of day and another array with the day opening time
      const dataScheduleValues = Object.entries(dataScheduleResponse);

      const daysWeek: DayRestaurantWeek[] = [];

      // This is looping through the array that contains days of the week
      dataScheduleValues.map((itemDay: [string, Array<HourDay>], index) => {
        // First position name of week
        const day = itemDay[0];

        // This second position an array with the opening time object of the day
        const openTime = itemDay[1];

        // This variable we will obtain if a restaurant has an opening time but no closing time
        let openingRestaurant = 0;

        // It is used to know if the restaurant is closed on that day
        if (openTime.length === 0) {
          daysWeek.push({
            day,
            dayTime: {
              open: 0,
              closed: 0,
            },
          });

          return;
        }

        // This is looping through the array that contains the opening time of each day
        openTime.forEach((item: HourDay) => {
          // This is to check if you the previous day time has an opening hour.
          if (openingRestaurant === 0) {
            if (item.type === 'open') {
              openingRestaurant = item.value || 0;
            }
          } else {
            if (item.type === 'close') {
              daysWeek.push({
                day,
                dayTime: {
                  open: openingRestaurant,
                  closed: item.value || 0,
                },
              });

              // We have to define a variable with zero here because daytime objects
              // that have closed property always have open time too
              openingRestaurant = 0;
            }
          }
        });

        // This is checking if some opening time doesn't have a closing time
        // And if dont have the next day time object is going to have the close time for that day
        if (openingRestaurant !== 0) {
          // Get the next day
          const nextDay = dataScheduleValues[index + 1];

          // Get first time of the next day
          // The index 1 is the array of daytime objects
          // And the index 0 is the first daytime
          const nextDayTime = nextDay[1][0];

          const nextDayFirstTime = nextDayTime.value;

          daysWeek.push({
            day,
            dayTime: {
              open: openingRestaurant,
              closed: nextDayFirstTime,
            },
          });
        }
      });

      return daysWeek;
    },
    []
  );

  const handleTransformSecondsHour = (timeSeconds: number) => {
    // It is get the hours using seconds
    const hours = Math.floor(timeSeconds / 3600);

    // This is defining the period using the 12 hours
    const period = hours >= 12 ? 'PM' : 'AM';

    // It is changing the hours from 24 format to 12 hours format
    const formattedHours = hours % 12 === 0 ? 12 : hours % 12;

    // This is used when the restaurant is closed
    if (timeSeconds === 0) {
      return String(0);
    }

    return `${formattedHours.toString()} ${period}`;
  };

  const handleDaysMoreOpenTime = (
    daysWeekDescription: DayRestaurantWeekString[]
  ) => {
    const mergedObjects: DayRestaurantWeekFormat[] = [];

    daysWeekDescription.forEach((item) => {
      const { day, dayTime } = item;

      // This findIndex will check if each day of the daysWeekDescription
      // array is in the mergedObjects array if not it is going
      // to add the day transforming the dayTime object in an array
      const existingIndex = mergedObjects.findIndex((obj) => obj.day === day);

      // In this line if the index is different from -1 that day
      // object already exists in the merged objects so add the dayTime
      if (existingIndex !== -1) {
        mergedObjects[existingIndex].dayTime.push(dayTime);
      } else {
        // In this line we are creating an object with day and dayTime in mergedObjects
        // Then if it returns -1 it will pass here and it means that need
        //Â to create the object
        mergedObjects.push({ day, dayTime: [dayTime] });
      }
    });

    return mergedObjects;
  };

  const handleScheduleItemDescription = useCallback(
    (daysWeek: DayRestaurantWeek[]) => {
      // This is transforming the seconds to day hour and handle when have day time closed
      const daysWeekDescription = daysWeek.map((item) => {
        if (item.dayTime.open === 0 && item.dayTime.closed === 0) {
          return {
            ...item,
            dayTime: {
              open: 'Closed',
              closed: 'Closed',
            },
          };
        }

        return {
          ...item,
          dayTime: {
            open: handleTransformSecondsHour(item.dayTime.open),
            closed: handleTransformSecondsHour(item.dayTime.closed),
          },
        };
      });

      // In the daysWeekDescription array when we have more than one opening time in day
      // it is repetaing the day with each day time so this function is going to remove
      // the repeated days and put one position for each day with an array of day time to
      // make easy to display this data
      const daysWeekFormat = handleDaysMoreOpenTime(daysWeekDescription);

      setHourRestaurantData(daysWeekFormat);
    },
    []
  );

  useEffect(() => {
    if (data && isSuccess && !isLoading) {
      const daysWeek = handleScheduleRestaurantData(data);

      handleScheduleItemDescription(daysWeek);
    }
  }, [
    data,
    isSuccess,
    isLoading,
    handleScheduleRestaurantData,
    handleScheduleItemDescription,
  ]);

  return {
    hourRestaurantData,
    error: isError,
    isLoading,
  };
};
