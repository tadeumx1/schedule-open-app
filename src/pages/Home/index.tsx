import React from 'react';

import { useHome } from './useHome';

import { Container, TitleErrorDescription, LoadingHourData } from './styles';
import HourCard from '../../components/HourCard';

const Home = () => {
  const { hourRestaurantData, isLoading, error, isSuccess } = useHome();

  return (
    <Container>
      {isLoading && <LoadingHourData testID="loading-restaurant-hour" />}
      {error && (
        <TitleErrorDescription>Something wrong occurred</TitleErrorDescription>
      )}
      {isSuccess && <HourCard hourRestaurantData={hourRestaurantData} />}
    </Container>
  );
};

export default Home;
