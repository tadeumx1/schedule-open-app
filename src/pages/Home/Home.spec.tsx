import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import { useGetRestaurantScheduleData } from '../../services/modules/time-restaurant/actions';
import Home from './index';

const hourRestaurantOpeningTimesResponse = {
  monday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  tuesday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  wednesday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  thursday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  friday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  saturday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  sunday: [
    { type: 'open', value: 43200 },
    { type: 'close', value: 75600 },
  ],
};

const hourRestaurantOpenCloseAnotherDayTimeResponse = {
  monday: [],
  tuesday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  wednesday: [],
  thursday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  friday: [{ type: 'open', value: 36000 }],
  saturday: [
    { type: 'close', value: 3600 },
    { type: 'open', value: 36000 },
  ],
  sunday: [
    { type: 'close', value: 3600 },
    { type: 'open', value: 43200 },
    { type: 'close', value: 75600 },
  ],
};

const hourRestaurantOpenCloseMoreTimesDayResponse = {
  monday: [],
  tuesday: [],
  wednesday: [],
  thursday: [
    {
      type: 'open',
      value: 36000,
    },
    {
      type: 'close',
      value: 64800,
    },
  ],
  friday: [{ type: 'open', value: 64800 }],
  saturday: [
    { type: 'close', value: 3600 },
    { type: 'open', value: 32400 },
    { type: 'close', value: 39600 },
    { type: 'open', value: 57600 },
    { type: 'close', value: 82800 },
  ],
  sunday: [
    {
      type: 'open',
      value: 43200,
    },
    {
      type: 'close',
      value: 75600,
    },
  ],
};

const hourRestaurantOpenCloseMoreTimesAnotherDayResponse = {
  monday: [],
  tuesday: [],
  wednesday: [],
  thursday: [
    { type: 'open', value: 32400 },
    { type: 'close', value: 39600 },
    { type: 'open', value: 54000 },
    { type: 'close', value: 82800 },
  ],
  friday: [{ type: 'open', value: 64800 }],
  saturday: [
    { type: 'close', value: 3600 },
    { type: 'open', value: 32400 },
    { type: 'close', value: 39600 },
    { type: 'open', value: 61200 },
    { type: 'close', value: 82800 },
  ],
  sunday: [
    {
      type: 'open',
      value: 43200,
    },
    {
      type: 'close',
      value: 75600,
    },
  ],
};

jest.mock('../../services/modules/time-restaurant/actions');

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

jest.useFakeTimers().setSystemTime(new Date('2023-05-16'));

const mockUseGetRestaurantScheduleData =
  useGetRestaurantScheduleData as jest.Mock;

describe('Home Component', () => {
  it('should render screen correct', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpeningTimesResponse,
        isError: false,
        isLoading: false,
        isSuccess: true,
      };
    });

    await waitFor(() => {
      const component = render(<Home />);

      expect(component).toMatchSnapshot();
    });
  });

  it('should render opening time restaurant when return data', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpeningTimesResponse,
        isError: false,
        isLoading: false,
        isSuccess: true,
      };
    });

    const { getByText, getAllByText } = render(<Home />);

    await waitFor(() => {
      expect(getByText('Monday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[0]).toBeDefined();
      expect(getByText('TODAY')).toBeDefined();
      expect(getByText('Tuesday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[1]).toBeDefined();
      expect(getByText('Wednesday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[2]).toBeDefined();
      expect(getByText('Thursday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[3]).toBeDefined();
      expect(getByText('Friday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[4]).toBeDefined();
      expect(getByText('Saturday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[5]).toBeDefined();
      expect(getByText('Sunday')).toBeDefined();
      expect(getByText('12 PM - 9 PM')).toBeDefined();
    });
  });

  it('should render error message when opening time restaurant when return error data', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpeningTimesResponse,
        isError: true,
        isLoading: false,
        isSuccess: false,
      };
    });

    const { getByText } = render(<Home />);

    await waitFor(() => {
      expect(getByText('Something wrong occurred')).toBeDefined();
    });
  });

  it('should render loading when opening time restaurant return data', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpeningTimesResponse,
        isError: false,
        isLoading: true,
        isSuccess: false,
      };
    });

    const { getByTestId } = render(<Home />);

    await waitFor(() => {
      expect(getByTestId('loading-restaurant-hour')).toBeDefined();
    });
  });

  it('should render opening time when restaurant open and close in another day return data', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpenCloseAnotherDayTimeResponse,
        isError: false,
        isLoading: false,
        isSuccess: true,
      };
    });

    const { getByText, getAllByText } = render(<Home />);

    await waitFor(() => {
      expect(getByText('Monday')).toBeDefined();
      expect(getAllByText('Closed')[0]).toBeDefined();
      expect(getByText('TODAY')).toBeDefined();
      expect(getByText('Tuesday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[0]).toBeDefined();
      expect(getByText('Wednesday')).toBeDefined();
      expect(getAllByText('Closed')[1]).toBeDefined();
      expect(getByText('Thursday')).toBeDefined();
      expect(getAllByText('10 AM - 6 PM')[1]).toBeDefined();
      expect(getByText('Friday')).toBeDefined();
      expect(getAllByText('10 AM - 1 AM')[0]).toBeDefined();
      expect(getByText('Saturday')).toBeDefined();
      expect(getAllByText('10 AM - 1 AM')[1]).toBeDefined();
      expect(getByText('Sunday')).toBeDefined();
      expect(getByText('12 PM - 9 PM')).toBeDefined();
    });
  });

  it('should render opening time when restaurant open and close more times in day return data', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpenCloseMoreTimesDayResponse,
        isError: false,
        isLoading: false,
        isSuccess: true,
      };
    });

    const { getByText, getAllByText } = render(<Home />);

    await waitFor(() => {
      expect(getByText('Monday')).toBeDefined();
      expect(getAllByText('Closed')[0]).toBeDefined();
      expect(getByText('TODAY')).toBeDefined();
      expect(getByText('Tuesday')).toBeDefined();
      expect(getAllByText('Closed')[1]).toBeDefined();
      expect(getByText('Wednesday')).toBeDefined();
      expect(getAllByText('Closed')[2]).toBeDefined();
      expect(getByText('Thursday')).toBeDefined();
      expect(getByText('10 AM - 6 PM')).toBeDefined();
      expect(getByText('Friday')).toBeDefined();
      expect(getByText('6 PM - 1 AM')).toBeDefined();
      expect(getByText('Saturday')).toBeDefined();
      expect(getByText('9 AM - 11 AM, 4 PM - 11 PM')).toBeDefined();
      expect(getByText('Sunday')).toBeDefined();
      expect(getByText('12 PM - 9 PM')).toBeDefined();
    });
  });

  it('should render opening time when restaurant open and close more times in many days return data', async () => {
    mockUseGetRestaurantScheduleData.mockImplementation(() => {
      return {
        data: hourRestaurantOpenCloseMoreTimesAnotherDayResponse,
        isError: false,
        isLoading: false,
        isSuccess: true,
      };
    });

    const { getByText, getAllByText } = render(<Home />);

    await waitFor(() => {
      expect(getByText('Monday')).toBeDefined();
      expect(getAllByText('Closed')[0]).toBeDefined();
      expect(getByText('TODAY')).toBeDefined();
      expect(getByText('Tuesday')).toBeDefined();
      expect(getAllByText('Closed')[1]).toBeDefined();
      expect(getByText('Wednesday')).toBeDefined();
      expect(getAllByText('Closed')[2]).toBeDefined();
      expect(getByText('Thursday')).toBeDefined();
      expect(getByText('9 AM - 11 AM, 3 PM - 11 PM')).toBeDefined();
      expect(getByText('Friday')).toBeDefined();
      expect(getByText('6 PM - 1 AM')).toBeDefined();
      expect(getByText('Saturday')).toBeDefined();
      expect(getByText('9 AM - 11 AM, 5 PM - 11 PM')).toBeDefined();
      expect(getByText('Sunday')).toBeDefined();
      expect(getByText('12 PM - 9 PM')).toBeDefined();
    });
  });
});
