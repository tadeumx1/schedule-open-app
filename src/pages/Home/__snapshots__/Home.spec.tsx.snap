// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render screen correct 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#eeeeee",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "elevation": 5,
          "maxWidth": 350,
          "minWidth": 300,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.7,
          "shadowRadius": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoBold",
              "fontSize": 24,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
            },
          ]
        }
      >
        Opening hours
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 0.5,
            "borderColor": "#202125",
            "marginBottom": 5,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#202125",
                "fontFamily": "RobotoMedium",
                "fontSize": 16,
                "marginRight": 10,
              },
            ]
          }
        >
          Monday
        </Text>
        <Text
          style={
            [
              {
                "color": "#5bcb02",
                "fontFamily": "RobotoBold",
                "fontSize": 12,
              },
            ]
          }
        >
          TODAY
        </Text>
      </View>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        10 AM - 6 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoMedium",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        Tuesday
      </Text>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        10 AM - 6 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoMedium",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        Wednesday
      </Text>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        10 AM - 6 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoMedium",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        Thursday
      </Text>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        10 AM - 6 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoMedium",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        Friday
      </Text>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        10 AM - 6 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoMedium",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        Saturday
      </Text>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        10 AM - 6 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "RobotoMedium",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        Sunday
      </Text>
      <Text
        closed={false}
        style={
          [
            {
              "color": "#202125",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 10,
            },
          ]
        }
      >
        12 PM - 9 PM
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaView>
`;
